### PR TITLE
Suggestion for PR #52

### PR DESCRIPTION
**Pull request by tarakby**
suggestions for #52 

- extend KMAC128 tests with an output size larger than 32 and an expected test vector
- update KMAC128 output size in the hashing algos list from 128 to 256
- update `HasherImpl` to enforce keys are customizers are only used for KMAC128
- added tests for more invalid inputs
  - non-nil keys for non-KMAC algos
  - non-nil customizers for non-KMAC algos
  - invalid output size for non-KMAC algos
  - non-supported hashing algo

### CODE REVIEW

File: src/main/kotlin/org/onflow/flow/sdk/models.kt
Comment (franklywatson) [inline src/main/kotlin/org/onflow/flow/sdk/models.kt:1]: When I run this using './gradlew test -i' I get 

```CryptoTest > Test output size longer than 32 bytes() FAILED
    org.opentest4j.AssertionFailedError: array contents differ at index [0], expected: <45> but was: <32>
        at app//org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
        at app//org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
        at app//org.junit.jupiter.api.AssertArrayEquals.failArraysNotEqual(AssertArrayEquals.java:440)
        at app//org.junit.jupiter.api.AssertArrayEquals.assertArrayEquals(AssertArrayEquals.java:209)
        at app//org.junit.jupiter.api.AssertArrayEquals.assertArrayEquals(AssertArrayEquals.java:63)
        at app//org.junit.jupiter.api.AssertArrayEquals.assertArrayEquals(AssertArrayEquals.java:59)
        at app//org.junit.jupiter.api.Assertions.assertArrayEquals(Assertions.java:1229)
        at app//org.onflow.flow.sdk.crypto.CryptoTest.Test output size longer than 32 bytes(CryptoTest.kt:244)```

File: src/test/kotlin/org/onflow/flow/sdk/crypto/CryptoTest.kt
Comment (franklywatson) [inline src/test/kotlin/org/onflow/flow/sdk/crypto/CryptoTest.kt:1]: When I run this using './gradlew test -i' I get 

```
CryptoTest > Test output size longer than 32 bytes() FAILED
    org.opentest4j.AssertionFailedError: array contents differ at index [0], expected: <45> but was: <32>
        at app//org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
        at app//org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
        at app//org.junit.jupiter.api.AssertArrayEquals.failArraysNotEqual(AssertArrayEquals.java:440)
        at app//org.junit.jupiter.api.AssertArrayEquals.assertArrayEquals(AssertArrayEquals.java:209)
        at app//org.junit.jupiter.api.AssertArrayEquals.assertArrayEquals(AssertArrayEquals.java:63)
        at app//org.junit.jupiter.api.AssertArrayEquals.assertArrayEquals(AssertArrayEquals.java:59)
        at app//org.junit.jupiter.api.Assertions.assertArrayEquals(Assertions.java:1229)
        at app//org.onflow.flow.sdk.crypto.CryptoTest.Test output size longer than 32 bytes(CryptoTest.kt:244)```

Comment (tarakby) [reply src/test/kotlin/org/onflow/flow/sdk/crypto/CryptoTest.kt:1]: That's the previous failing test with an incompatible test vector. I removed it in a new commit

**Review by franklywatson — APPROVED** [review]

**Review by franklywatson — APPROVED** [review]

**Comment by lealobanov** [comment]
@tarakby just had a look at the changes and I think everything looks good, I am rebasing against main to account for all the changed file paths and fixed the failing build due to linter. As soon as build passes will go ahead and merge this, thanks!

**Comment by lealobanov** [comment]
@tarakby @franklywatson There is currently one failing test against the emulator which seems to be coming from the Cadence index, so will go ahead and merge this and resolve in #52 . Thanks!

**Pull request merged**